test/integration/apiserver/apply/reset_fields_test.go

```diff
@@ -268,24 +269,28 @@ func TestApplyResetFields(t *testing.T) {
 				// skip checking for conflicts on resources
 				// that will never have conflicts
 				if _, ok = noConflicts[mapping.Resource.Resource]; !ok {
+					var objRet *unstructured.Unstructured
+
 					// reapply second object to the spec endpoint
 					// that should fail with a conflict
-					_, err = dynamicClient.
+					objRet, err = dynamicClient.
 						Resource(mapping.Resource).
 						Namespace(namespace).
 						Apply(context.TODO(), name, obj2, metav1.ApplyOptions{FieldManager: "fieldmanager2"})
-					if err == nil || !strings.Contains(err.Error(), "conflict") {
-						t.Fatalf("expected conflict, got error %v", err)
+					err = expectConflict(objRet, err, dynamicClient, mapping.Resource, namespace, name)
+					if err != nil {
+						t.Fatalf("Did not get expected conflict in spec of %s %s/%s: %v", mapping.Resource, namespace, name, err)
```

Comment:
instead of passing `dynamicClient, mapping.Resource, namespace`, you can pass `dynamicClientResource(resource).Namespace(namespace)`?

Reply:
I would prefer to not break up the `dynamicClient.Resource(resource).Namespace(namespace).Get(..)` inside expectConflict, I would prefer to keep the parallel structure with the other API calls.